Added MS Face APIs

bruden01 opened the pull request:
This is a set of Microsoft Face API's from Azure Cognitive Services.  It provides a series of functions for performing face detection (gender, emotion, accessories, etc...), face recognition, as well as training and grouping.

bruden01 committed:
Added MS Face APIs

rblumer4 commented on MSFaceAPIs/projects/MSFaceAPIs.json:
```diff
@@ -0,0 +1,546 @@
+  } ],
+  "name" : "MSFaceAPIs",
+  "options" : {
+    "description" : "MICROSOFT FACE API'S\n\nIn order to use these API's you will need to set up the Azure FaceAPI and obtain a subscription key.\n\nThe modify the faceapi source in VANTIQ and set the key\n{\"Ocp-Apim-Subscription-Key\":\"<Your Subscription Key>\"}\n\nIf all is setup correctly you should be able to run the 3 procedures to the right in order\n\nFor more info on the face api visit: ...
```
Typos:  The -> Then, that -> since
```suggestion
     "description" : "MICROSOFT FACE API'S\n\nIn order to use these API's you will need to set up the Azure FaceAPI and obtain a subscription key.\n\nThen modify the faceapi source in VANTIQ and set the key\n{\"Ocp-Apim-Subscription-Key\":\"<Your Subscription Key>\"}\n\nIf all is setup correctly you should be able to run the 3 procedures to the right in order\n\nFor more info on the face api visit:\n\nhttps://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395236\n\nThis is a collection of API connection to Azure cognitive services.\n\nSpecifically the FACE API\n\nAPI's\n-------------------------\nFaceDetect:\nFind faces in an image and return attributes (age, gender, facialHair, glasses, emotion, hair, makeup, accessories)\n\nFaceIdentify:\nFor a face found in FaceDetect see if there are any matches against a particular person group. (collection of people and faces)\n\nFaceListList:\nList faces added to the facelist\n\nPersonGroupCreate:\nCreate a person group\n\nPersonGroupDelete:\nDelete a person group\n\nPersonGroupGetTrainingStatus:\nGet the training status of a person group\n\nPersonGroupList:\nList all person groups\n\nPersonGroupPersonCreate:\nCreate a person in a person group\n\nPersonGroupPersonAddFace:\nAdd a face to a person in the person group\n\n\nPersonGroupPersonDelete:\nDelete a person from the person group\n\nPersonGroupPersonGet:\nGet the details of a person in a person group\n\nPersonGroupPersonGetFace:\nRetrieve person face information. The persisted person face is specified by its personGroupId, personId and persistedFaceId.\n\nPersonGroupPersonList:\nList all persons’ information in the specified person group, including personId, name, userData and persistedFaceIds of registered person faces.\n\nPersonGroupTrain:\nSubmit a person group training task. Training is a crucial step since only a trained person group can be used by Face - Identify. \n\n\n\n\n\n",
```

JakeSchmitz reviewed (changes requested):
One request for consistency. Can you capitalize the first letter of the service name in each procedure, and lowercase the first letter of the procedure name? This is consistent with all of the built ins VANTIQ has added. E.g: msFaceAPI.FaceDetect -> MSFaceAPI.faceDetect

bruden01 committed:
Fixed prcedured names per Jake naming convention

bruden01 commented:
Ok..  Changes to procedures have been made per @JakeSchmitz

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.createGroupAndPerson.vail:
```diff
@@ -0,0 +1,38 @@
+	obj.faceRectangle.height
+)
+
+var faceResult = msFaceAPI.PersonGroupPersonAddFace("friend_group", obj.personId, "https://cdn.pixabay.com/photo/2016/08/24/14/32/man-1617126_1280.jpg", "Trip to the mountains", targetFace)
```
This is now using the wrong procedure/ service name. Looks like there are a lot of these references that need to be fixed now that you've changed the capitalization of the procedure and service names

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.createGroupAndPerson.vail:
```diff
@@ -0,0 +1,38 @@
+obj.persistedFaceId = faceResult[0].body.persistedFaceId
+
+// train the face(s) in the person group
+msFaceAPI.PersonGroupTrain("friend_group")
```
here

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.identifyFace.vail:
```diff
@@ -0,0 +1,23 @@
+PROCEDURE FaceDemo.identifyFace()
+
+// in order to identify a face we first need to detect faces
+var faceResults = msFaceAPI.FaceDetect("https://cdn.pixabay.com/photo/2016/08/24/14/32/man-1617126_1280.jpg")
```
here

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.identifyFace.vail:
```diff
@@ -0,0 +1,23 @@
+
+var matches = []
+FOR EACH item IN candidates DO
+	var person = msFaceAPI.PersonGroupPersonGet("friend_group", item.personId )
```
here

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.createGroupAndPerson.vail:
```diff
@@ -0,0 +1,38 @@
+var obj= {}
+
+// create a new person group
+msFaceAPI.PersonGroupCreate("friend_group", "Friends", "Person group identifies friends")
```
here

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.createGroupAndPerson.vail:
```diff
@@ -0,0 +1,38 @@
+
+var personResult = msFaceAPI.PersonGroupPersonCreate("friend_group", "Joe Smith", "Met him while hiking")
+
+obj.personId = personResult[0].body.personId
```
here

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.createGroupAndPerson.vail:
```diff
@@ -0,0 +1,38 @@
+
+obj.personId = personResult[0].body.personId
+
+var faceResults = msFaceAPI.FaceDetect("https://cdn.pixabay.com/photo/2016/08/24/14/32/man-1617126_1280.jpg")
```
here

JakeSchmitz commented on MSFaceAPIs/procedures/facedemo.resetDemo.vail:
```diff
@@ -0,0 +1,4 @@
+PROCEDURE FaceDemo.resetDemo()
+
+msFaceAPI.PersonGroupDelete("friend_group")
```
here

JakeSchmitz reviewed (changes requested):
Marked some spots where names need to be fixed to reflect capitalization changes

bruden01 committed:
fixed demo procedures that call face APIs

bruden01 commented:
Ok. I think I fixed the remaining issues with the demo procedures not pointing to the right API's after I changed the case of the procedure names. Let me know if there are any other issues.

bruden01 committed:
Added author so people would know who to contact with questions

bruden01 commented:
is this ready to go or am I still missing something?

JakeSchmitz commented:
One last issue is the file names no longer match the procedure names. Sorry I didn't make that clear in my last comment.

bruden01 commented:
how do I fix that? The export procedure names them.  Or do you mean
manually?

On Thu, Jan 3, 2019 at 8:14 PM Jacob Schmitz <notifications@github.com>
wrote:

> One last issue is the file names no longer match the procedure names.
> Sorry I didn't make that clear in my last comment.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Vantiq/Contributions/pull/3#issuecomment-451329022>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACbhr7UwOx7jNd0kd-Kn9FsyAG4zTbd7ks5u_qsEgaJpZM4ZcJjT>
> .
>

rblumer4 commented:
If you re-export the procedures after the rename, the files should get the new name.
If you renamed them in the existing export directory, rather than in your namespace, you could import the changed files & then re-export them and the procedure files should get the new names...
To get them renamed in GIT, you'd have to manually delete the old files (_git rm_), do the export, then add the new files (_git add_) and commit them.

bruden01 committed:
Fixed file system name capitalization

bruden01 committed:
Rename facedemo.createGroupAndPerson.vail to FaceDemo.createGroupAndPerson.vail

bruden01 committed:
Rename facedemo.identifyFace.vail to FaceDemo.identifyFace.vail

bruden01 committed:
Rename facedemo.resetDemo.vail to FaceDemo.resetDemo.vail

bruden01 committed:
Rename msFaceAPI.FaceDetect.vail to MSFaceAPI.faceDetect.vail

bruden01 committed:
Rename msFaceAPI.FaceIdentify.vail to MSFaceAPI.faceIdentify.vail

bruden01 committed:
Rename msFaceAPI.FaceListList.vail to MSFaceAPI.faceListList.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupCreate.vail to MSFaceAPI.personGroupCreate.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupDelete.vail to MSFaceAPI.personGroupDelete.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupGetTrainingStatus.vail to MSFaceAPI.personGroupGetTrainingStatus.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupList.vail to MSFaceAPI.personGroupList.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupPersonAddFace.vail to MSFaceAPI.personGroupPersonAddFace.vail

pburma committed:
Rename msFaceAPI.PersonGroupTrain.vail to MSFaceAPI.personGroupTrain.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupPersonCreate.vail to MSFaceAPI.personGroupPersonCreate.vail

pburma committed:
Rename msFaceAPI.PersonGroupPersonList.vail to MSFaceAPI.personGroupPersonList.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupPersonDelete.vail to MSFaceAPI.personGroupPersonDelete.vail

bruden01 committed:
Rename msFaceAPI.PersonGroupPersonGet.vail to MSFaceAPI.personGroupPersonGet.vail

pburma committed:
Rename msFaceAPI.PersonGroupPersonGetFace.vail to MSFaceAPI.personGroupPersonGetFace.vail

bruden01 commented:
I think this is ready now.

JakeSchmitz reviewed (approved):
LGTM
